libs/api/domains/national-registry/src/lib/nationalRegistry.service.ts

```diff
@@ -26,7 +26,15 @@ export class NationalRegistryService {
     return this.nationalRegistryApi.getBanMarking(nationalId)
   }
 
-  getLegalResidence(houseCode: User['houseCode']): Promise<string> {
-    return this.nationalRegistryApi.getLegalResidence(houseCode)
+  async getLegalResidence(houseCode: User['houseCode']): Promise<string> {
+    const address = await this.nationalRegistryApi.getAddress(houseCode)
+    if (!address) {
+      return ''
+    }
+    return `${address.streetAddress}, ${address.postalCode} ${address.city}`
+  }
+
+  getAddress(houseCode: User['houseCode']) {
+    return this.nationalRegistryApi.getAddress(houseCode)
```

Comment:
Did you have a look at if the soap client is caching anything (wondering about two requests made)